ciphers.go

```diff
@@ -287,6 +287,9 @@ func NewDecryptionCipherCtx(c *Cipher, e *Engine, key, iv []byte) (
 }
 
 func (ctx *encryptionCipherCtx) EncryptUpdate(input []byte) ([]byte, error) {
+	if input == nil {
```

Comment:
I think this actually needs to check for `len(input) == 0`. In other words, passing `[]byte{}` will also panic.
```suggestion
	if len(input) == 0 {
```

I don't actually know what the semantics should be with a zero sized buffer, though. Like, is that even a valid thing one would want to pass?

Reply:
I believe check for `len(input) == 0` will work for both `nil` and `empty` data. Let me change that and update the PR.

Reply:
fixed in the new commit. Thanks !